src/delagent/ui/delete-file-browse.php

```diff
@@ -0,0 +1,149 @@
+use Fossology\DelAgent\UI\DeleteMessages;
+use Fossology\DelAgent\UI\DeleteResponse;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+
+/**
+ * @class BrowseUploadDelete
+ * @brief UI plugin to delete uploaded files
+ */
+class BrowseUploadDelete extends DefaultPlugin
+{
+  const NAME = "browse_upload_delete";
+
+  /** @var UploadDao */
+  private $uploadDao;
+
+  /** @var FolderDao */
+  private $folderDao;
+
+  function __construct()
+  {
+    parent::__construct(self::NAME, array(
+        self::TITLE => _("Delete Uploaded File"),
+        self::PERMISSION => Auth::PERM_ADMIN,
+        self::REQUIRES_LOGIN => true
+    ));
+
+    global $container;
+    $this->uploadDao = $container->get('dao.upload');
+    $this->folderDao = $container->get('dao.folder');
+  }
+
+    /**
+   * \brief Customize submenus.
+   */
+  function RegisterMenus()
+  {
+    $text = _("Delete Browse File");
+    menu_insert("Browse-Pfile::Delete",0,$this->Name,$text);
+  } // RegisterMenus()
+
+
+  /**
+   * @brief Delete a given upload
+   * @param int $uploadpk The upload(upload_id) you want to delete
+   * @return NULL on success, string on failure.
+   */
+  private function delete($uploadpk)
+  {
+    /* Prepare the job: job "Delete" */
+    $user_pk = Auth::getUserId();
+    $group_pk = Auth::getGroupId();
+    $jobpk = JobAddJob($user_pk, $group_pk, "Delete", $uploadpk);
+    if (empty($jobpk) || ($jobpk < 0)) {
+      return _("Failed to create job record");
+    }
+    /* Add job: job "Delete" has jobqueue item "delagent" */
+    $jqargs = "DELETE UPLOAD $uploadpk";
+    $jobqueuepk = JobQueueAdd($jobpk, "delagent", $jqargs, null, null);
+    if (empty($jobqueuepk)) {
+      return _("Failed to place delete in job queue");
+    }
+
+    /* Tell the scheduler to check the queue. */
+    $success  = fo_communicate_with_scheduler("database", $output, $error_msg);
+    if (!$success) {
+      $error_msg = _("Is the scheduler running? Your jobs have been added to job queue.");
+      $URL = Traceback_uri() . "?mod=showjobs&upload=$uploadpk ";
+      $LinkText = _("View Jobs");
+      return "$error_msg <a href=\"$URL\">$LinkText</a>";
+    }
+    return null;
+  }
+
+  /**
+   * @copydoc Fossology::Lib::Plugin::DefaultPlugin::handle()
+   * @see Fossology::Lib::Plugin::DefaultPlugin::handle()
+   */
+  protected function handle(Request $request)
+  {
+    $vars = array();
+
+    $uploadpk = $request->get('upload');
+    $folderId = $request->get('folder');
+
+    if (!empty($uploadpk)) {
+      $vars['message'] = $this->TryToDelete($uploadpk, $folderId);
+    }
+
+
+    return new RedirectResponse(Traceback_uri() . '?mod=' . 'browse');
+  }
+  
+  /**
+   * @brief Given a folder_pk, try to add a job after checking permissions.
+   * @param $uploadpk The upload(upload_id) you want to delete
+   * @param $folderId The folder(folder_id) containing the uploads
+   * @return string with the message.
+   */
+  private function TryToDelete($uploadpk, $folderId)
```

Comment:
Would recommend using the original function from following class instead of duplicating the code.
https://github.com/fossology/fossology/blob/416a40a84029d33c8cd715ecc56d44e571e5aa89/src/delagent/ui/Page/AdminUploadDelete.php#L171-L198

It can be access by making the function public and importing/calling it as bellow:
```php
/**
 * @var AdminUploadDelete $adminUploadDelete
 */
$adminUploadDelete = plugin_find("admin_upload_delete");
$adminUploadDelete->TryToDelete($uploadpk, $folderId);
```

Reply:
Okay sir got it I will work on the changes and update my pr soon